# Demonstrando a Importância da Vectorização (Vectorization)

## Comparando duas formas de vectorização (uma sem utilizar numpy e outra utilizando numpy)

In [2]:
a = np.array([1,2,3,4])

In [3]:
print(a)

[1 2 3 4]


In [4]:
import time
a = np.random.rand(1000000)

In [14]:
b = np.random.rand(1000000)
tic  = time.time()
c = np.dot(a, b)
toc = time.time()

print(c)
print("Vectorized version = " + str(1000*(toc-tic)) + " ms")

250088.652244886
Vectorized version = 1.4297962188720703 ms


In [17]:
c = 0
tic = time.time()
for i in range(1000000):
    c += a[i]*b[i]
toc = time.time()

print(c)
print("For loop version: " + str(1000*(toc-tic)) + "ms")

250088.65224487864
For loop version: 716.3949012756348ms


Como podemos observar o tempo que levou para computar com vectorization (numpy) foi de 1.4297962188720703 ms e sem vectorization (sem numpy) foi de 716.3949012756348 ms